SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorCommandLineArgsManager.cpp

```diff
@@ -0,0 +1,303 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialGDKEditorCommandLineArgsManager.h"
+
+#include "Input/Reply.h"
+#include "IOSRuntimeSettings.h"
+#include "Logging/LogMacros.h"
+#include "Misc/FileHelper.h"
+#include "Misc/MessageDialog.h"
+#include "Serialization/JsonSerializer.h"
+#include "SpatialCommandUtils.h"
+#include "SpatialGDKEditorSettings.h"
+#include "SpatialGDKSettings.h"
+
+#ifdef ENABLE_LAUNCHER_DELEGATE
+//need this macro before "ILauncherServicesModule.h" to prevent compile error
+#define LAUNCHERSERVICES_API
+
+#include "Developer/LauncherServices/Public/ILauncherServicesModule.h"
+#include "Developer/LauncherServices/Public/ILauncherWorker.h"
+#include "Developer/LauncherServices/Public/ILauncher.h"
+#endif
+
+DECLARE_LOG_CATEGORY_EXTERN(LogSpatialGDKEditorCommandLineArgsManager, Log, All);
+DEFINE_LOG_CATEGORY(LogSpatialGDKEditorCommandLineArgsManager);
+
+FSpatialGDKEditorCommandLineArgsManager::FSpatialGDKEditorCommandLineArgsManager()
+#ifdef ENABLE_LAUNCHER_DELEGATE
+	: bAndroidDevice(false)
+#endif
+{
+
+}
+
+void FSpatialGDKEditorCommandLineArgsManager::Startup()
+{
+#ifdef ENABLE_LAUNCHER_DELEGATE
+	ILauncherServicesModule& LauncherServicesModule = FModuleManager::LoadModuleChecked<ILauncherServicesModule>(TEXT("LauncherServices"));
+	LauncherServicesModule.OnCreateLauncherDelegate.AddRaw(this, &FSpatialGDKEditorCommandLineArgsManager::OnCreateLauncher);
+#endif
+}
+
+#ifdef ENABLE_LAUNCHER_DELEGATE
+void FSpatialGDKEditorCommandLineArgsManager::OnLauncherCanceled(double ExecutionTime)
+{
+	RemoveFromDevice();
+}
+
+void FSpatialGDKEditorCommandLineArgsManager::OnLauncherFinished(bool Outcome, double ExecutionTime, int32 ReturnCode)
+{
+	RemoveFromDevice();
+}
+
+void FSpatialGDKEditorCommandLineArgsManager::RemoveFromDevice()
+{
+	if (bAndroidDevice)
+	{
+		RemoveFromAndroidDevice();
+	}
+}
+
+void FSpatialGDKEditorCommandLineArgsManager::OnLaunch(ILauncherWorkerPtr LauncherWorkerPtr, ILauncherProfileRef LauncherProfileRef)
+{
+	LauncherWorkerPtr->OnCanceled().AddRaw(this, &FSpatialGDKEditorCommandLineArgsManager::OnLauncherCanceled);
+	LauncherWorkerPtr->OnCompleted().AddRaw(this, &FSpatialGDKEditorCommandLineArgsManager::OnLauncherFinished);
```

Comment:
I think that if I click launch on device multiple times, it will add this delegate also multiple times.
We either only do this once when starting up or we need to remove them whenever the launch on device session ends.